File: 3074-apple-redistribution-into-boxes/3074-apple-redistribution-into-boxes.cpp
class Solution {
public:
    int minimumBoxes(vector<int>& apple, vector<int>& capacity) {
        sort(capacity.begin(), capacity.end(), greater<>());
        int sum = accumulate(apple.begin(), apple.end(), 0);
        int i = 0;
        while(i < capacity.size() && sum > 0) {
            sum -= capacity[i];
            i++;
        }

        return i;
    }
};

File: 3074-apple-redistribution-into-boxes/3074-apple-redistribution-into-boxes.go
func minimumBoxes(apple []int, capacity []int) int {
    sort.Slice(capacity, func(i, j int) bool {
        return capacity[i] > capacity[j]
    })

    sum := 0
    for _, a := range apple {
        sum += a
    }

    i := 0
    for i < len(capacity) && sum > 0 {
        sum -= capacity[i]
        i++
    }

    return i
}